Mapping/Loader/ReflectionLoader.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/*
+ * This file is part of the DunglasApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\ApiBundle\Mapping\Loader;
+
+use Dunglas\ApiBundle\Mapping\Factory\AttributeMetadataFactoryInterface;
+use Dunglas\ApiBundle\Mapping\AttributeMetadataInterface;
+use Dunglas\ApiBundle\Mapping\ClassMetadataInterface;
+
+/**
+ * Uses serialization groups or alternatively reflection to populate attributes.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ReflectionLoader implements LoaderInterface
+{
+    const DEFAULT_IDENTIFIER = 'id';
+
+    /**
+     * @var AttributeMetadataFactoryInterface
+     */
+    private $attributeMetadataFactory;
+
+    public function __construct(AttributeMetadataFactoryInterface $attributeMetadataFactory)
+    {
+        $this->attributeMetadataFactory = $attributeMetadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadClassMetadata(
+        ClassMetadataInterface $classMetadata,
+        array $normalizationGroups = null,
+        array $denormalizationGroups = null,
+        array $validationGroups = null
+    ) {
+        if (null !== $normalizationGroups && null !== $denormalizationGroups) {
```

Comment:
I can't get the point here... If there's some normalisation groups and some denormalisation groups, then we skip any "reflection loading". Why that ?